src/TimePicker/TimePicker.tsx

```diff
@@ -41,6 +43,21 @@ const isValidHoursInTimeFormat = (
   timeFormat: TimePicker.TimeFormat
 ): hours is number => (timeFormat === H24 ? Hour.is(hours) : Hour12.is(hours));
 
+const getComponents = (
+  timeFormatter?: TimePicker.TimeFormatter
+): SelectNS.Props<TimeDropdownOption>["components"] => {
+  return timeFormatter
+    ? {
+        Option: props => (
+          <components.Option {...props}>
+            {timeFormatter(props.data.time) || null}
```

Comment:
why `|| null` if `timeFormatter` returns a `JSX.Element`?

Reply:
this is a very good question for which I have no reasonable answer...